### PR TITLE
Apple Pay

### PR DESCRIPTION
**Pull request by antiv0**
Initiating example of how to use Apple Pay JS and perform pay with Apple Pay against Circle endpoints.
Supported only in staging with this PR, and then to other environments.

### CODE REVIEW

File: server-middleware/applepay.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import axios from 'axios'
+import express from 'express'
+import * as https from 'https'
+import {
+  merchantIdentityCertificate,
+  merchantIdentityKey,
+} from './applePaySecrets'
```
Comment (antiv0) [inline server-middleware/applepay.ts:7]: Server middleware is loaded by Node, so used relative paths (couldn't use @/..)
 If there is a way to make it nicer ?

File: lib/applePay.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  'https://sample-staging.circle.com/applepay/validate'
+const BACKEND_URL_PAY = 'https://sample-staging.circle.com/applepay/pay'
+
+// default configuration used in staging
+const default_config = {
+  payments: {
+    acceptedCardSchemes: ['amex', 'masterCard', 'visa'],
+  },
+  shop: {
+    product_price: '0.5',
```
Comment (antiv0) [inline lib/applePay.ts:14]: Price $0.5 for testing hardcoded (didn't want to involve forms and stuff until we see that apple pay can work end to end + until we have UI expectations)

**Review by kristinfritsch — CHANGES_REQUESTED** [review]

**Comment by kristinfritsch** [comment]
If i call `http://localhost:3011/applepay/pay` i get page not found

File: server-middleware/applePaySecrets.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const applePaySecretsAreSet = !(
+  process.env.APPLE_PAY_CERTIFICATE == null ||
+  process.env.APPLE_PAY_PRIVATE_KEY == null
+)
+
+const merchantIdentityCertificate: string = applePaySecretsAreSet
+  ? process.env.apple_pay_certifcate!
```
[the remark this answers is not in the document]
Comment (antiv0) [reply server-middleware/applePaySecrets.ts:7]: Thank you !!

File: layouts/default.vue [hunk 4]
```diff
@@ -625,8 +639,15 @@ export default class DefaultLayoutsClass extends Vue {
   showRightDrawer = false
   showDrawer = false
 
+  getPaymentLinks() {
```
[the remark this answers is not in the document]
Comment (antiv0) [reply layouts/default.vue:642]: sure,
// i do get that `get` names a property/attribute, like in C#
// changing method to property

**Comment by antiv0** [comment]
> If i call `http://localhost:3011/applepay/pay` i get page not found, doesn't seem right

curl -X POST -H "Content-Type: application/json" http://192.168.1.6:3011/applepay/pay -d '{}'

File: lib/applePay.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
+import axios from 'axios'
+
+// endpoints are hardcoded as they are used only in staging
+const BACKEND_URL_VALIDATE_SESSION =
+  'https://sample-staging.circle.com/applepay/validate'
```
[the remark this answers is not in the document]
Comment (kristinfritsch) [reply lib/applePay.ts:5]: can it be configurable via env var

**Review by kristinfritsch — DISMISSED** [review]

**Review by kristinfritsch — APPROVED** [review]

**Pull request merged**